src/render.jl

```diff
@@ -5,28 +5,38 @@ parsedocs(ds::Docs{:md}) = Markdown.parse(data(ds))
 ## Common -------------------------------------------------------------------------------
 const MDHTAGS = ["#", "##", "###", "####", "#####", "######"]
 const MDSTYLETAGS = ["", "*", "**"]
+const MD_SUBHEADER_OPTIONS = [:SKIP, :SIMPLE, :CATEGORY]
 
 type Config
-    # General Options
-    include_internal :: Bool
-    # Html only Options
-    mathjax          :: Bool
-    # MarkDown only Options
-    mdstyle_header   :: ASCIIString
-    mdstyle_objname  :: ASCIIString
-    mdstyle_meta     :: ASCIIString
-    mdstyle_exported :: ASCIIString
-    mdstyle_internal :: ASCIIString
+    category_order         :: Vector{Symbol}
+    include_internal       :: Bool
+    mathjax                :: Bool
+    mdstyle_header         :: ASCIIString
+    mdstyle_objname        :: ASCIIString
+    mdstyle_meta           :: ASCIIString
+    mdstyle_subheader      :: ASCIIString
+    mdstyle_genindex_mod   :: ASCIIString
+    md_subheader           :: Symbol
+    md_genindex            :: Bool
+    md_genindex_modprefix  :: ByteString
+    md_permalink           :: Bool
 
     const fields   = fieldnames(Config)
+
     const defaults = Dict{Symbol, Any}([
-        (:include_internal , true),
-        (:mathjax          , false),
-        (:mdstyle_header   , "#"),
-        (:mdstyle_objname  , "###"),
-        (:mdstyle_meta     , "*"),
-        (:mdstyle_exported , "##"),
-        (:mdstyle_internal , "##")
+        (:category_order         , [:module, :function, :method, :type, :typealias, :macro, :global,
+                                                                                         :comment]),
```

Comment:
Hard wrap at

```julia
        (:category_order         , [:module, :function, :method, :type,
                                    :typealias, :macro, :global, :comment]),
```

General rule of thumb I use is to make the two lines as equal as possible when something needs to move to the next line and line up with the `[`.